test/cmdlineargs.jl

```diff
@@ -1182,24 +1186,29 @@ end
     end
 end
 
+# FIXME: MMTK currently does not use --heap-size-hint since it only
+# supports setting up a hard limit unlike the Stock GC
+# which takes it as a soft limit. For now, we skip the tests below for MMTk
```

Comment:
Same thing here.

Seems like you forgot the `@static` before the `if`.

Also, please guard the whole `@testset "heap size hint"`.